### ISSUE TITLE
训练时报错

### ISSUE THREAD

**Issue by friendiszq** [zepen/predict_Lottery_ticket#72]
请问一下训练了30组之后就提示有错误（Issue encountered when serializing variables），这个是正常的吗？如果不正常是什么原因导致出错的？

/root/anaconda3/envs/your_env_name/lib/python3.6/site-packages/tensorflow_addons/utils/ensure_tf_install.py:68: UserWarning: Tensorflow Addons supports using Python ops for all Tensorflow versions above or equal to 2.2.0 and strictly below 2.3.0 (nightly versions are not supported). 
 The versions of TensorFlow you are currently using is 2.6.1 and is not supported. 
Some things might work, some things might not.
If you were to encounter a bug, do not file an issue.
If you want to make sure you're using a tested and supported configuration, either change the TensorFlow version or the TensorFlow Addons's version. 
You can find the compatibility matrix in TensorFlow Addon's readme:
https://github.com/tensorflow/addons
  UserWarning,
2023-03-20 11:45:29.725 | INFO     | __main__:run:193 - 正在创建【双色球】数据集...
2023-03-20 11:45:29.736 | INFO     | __main__:create_train_data:34 - 训练数据已加载! 
2023-03-20 11:45:29.738 | INFO     | __main__:create_train_data:37 - 训练集数据维度: (2974, 7)
2023-03-20 11:45:29.749 | INFO     | __main__:run:196 - 开始训练【双色球】红球模型...
2023-03-20 11:45:29.750 | INFO     | __main__:train_red_ball_model:66 - 特征数据维度: (2970, 3, 6)
2023-03-20 11:45:29.750 | INFO     | __main__:train_red_ball_model:67 - 标签数据维度: (2970, 6)
2023-03-20 11:45:29.767505: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 AVX512F FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-03-20 11:45:39.426557: W tensorflow/c/c_api.cc:335] Operation '{name:'lstm_5/while' id:1542 op device:{requested: '', assigned: ''} def:{{{node lstm_5/while}} = While[T=[DT_INT32, DT_INT32, DT_INT32, DT_VARIANT, DT_FLOAT, DT_FLOAT, DT_INT32, DT_VARIANT, DT_RESOURCE, DT_RESOURCE, DT_RESOURCE, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT, DT_VARIANT], _lower_using_switch_merge=true, _num_original_outputs=44, _read_only_resource_inputs=[8, 9, 10], _stateful_parallelism=false, body=lstm_5_while_body_2407[], cond=lstm_5_while_cond_2406[], output_shapes=[[], [], [], [], [1,32], [1,32], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], [], []], parallel_iterations=32](lstm_5/while/loop_counter, lstm_5/while/maximum_iterations, lstm_5/time, lstm_5/TensorArrayV2_1, lstm_5/zeros, lstm_5/zeros_1, lstm_5/strided_slice_1, lstm_5/TensorArrayUnstack/TensorListFromTensor, lstm_5/lstm_cell_5/kernel, lstm_5/lstm_cell_5/recurrent_kernel, lstm_5/lstm_cell_5/bias, lstm_5/while/EmptyTensorList, lstm_5/while/EmptyTensorList_1, lstm_5/while/EmptyTensorList_2, lstm_5/while/EmptyTensorList_3, lstm_5/while/EmptyTensorList_4, lstm_5/while/EmptyTensorList_5, lstm_5/while/EmptyTensorList_6, lstm_5/while/EmptyTensorList_7, lstm_5/while/EmptyTensorList_8, lstm_5/while/EmptyTensorList_9, lstm_5/while/EmptyTensorList_10, lstm_5/while/EmptyTensorList_11, lstm_5/while/EmptyTensorList_12, lstm_5/while/EmptyTensorList_13, lstm_5/while/EmptyTensorList_14, lstm_5/while/EmptyTensorList_15, lstm_5/while/EmptyTensorList_16, lstm_5/while/EmptyTensorList_17, lstm_5/while/EmptyTensorList_18, lstm_5/while/EmptyTensorList_19, lstm_5/while/EmptyTensorList_20, lstm_5/while/EmptyTensorList_21, lstm_5/while/EmptyTensorList_22, lstm_5/while/EmptyTensorList_23, lstm_5/while/EmptyTensorList_24, lstm_5/while/EmptyTensorList_25, lstm_5/while/EmptyTensorList_26, lstm_5/while/EmptyTensorList_27, lstm_5/while/EmptyTensorList_28, lstm_5/while/EmptyTensorList_29, lstm_5/while/EmptyTensorList_30, lstm_5/while/EmptyTensorList_31, lstm_5/while/EmptyTensorList_32)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.
2023-03-20 11:45:42.942 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 21.34540367126465, tag: [ 2  3  7 16 30 31], pred: [19 23  1 11  3  6]
2023-03-20 11:45:43.969 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 20.08443832397461, tag: [ 5 12 15 17 18 27], pred: [ 7 18 26 30 31 31]
2023-03-20 11:45:45.355 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 17.232921600341797, tag: [ 8  9 15 24 26 30], pred: [ 2  5 12 18 20 31]
2023-03-20 11:45:46.172 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 17.593353271484375, tag: [ 8  9 13 27 29 31], pred: [ 1 11 17 22 24 29]
2023-03-20 11:45:47.035 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 14.672489166259766, tag: [ 1  3  7 21 27 32], pred: [ 1  9 10 17 25 30]
2023-03-20 11:45:47.888 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 16.827129364013672, tag: [ 4 10 11 14 15 29], pred: [ 1  9 10 17 27 33]
2023-03-20 11:45:48.744 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 17.6677303314209, tag: [11 15 16 20 24 31], pred: [ 3  6 14 22 23 33]
2023-03-20 11:45:49.608 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 16.01062774658203, tag: [ 8 10 17 20 27 30], pred: [ 1  6 14 22 23 33]
2023-03-20 11:45:50.506 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 18.00617218017578, tag: [11 20 21 22 24 27], pred: [ 1  7 18 20 26 33]
2023-03-20 11:45:51.374 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 14.791516304016113, tag: [ 2  6 15 16 18 32], pred: [ 1  7 14 22 23 31]
2023-03-20 11:45:52.312 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 14.343476295471191, tag: [ 1  3  6 16 29 32], pred: [ 1  7  8 17 27 33]
2023-03-20 11:45:53.243 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 15.958414077758789, tag: [ 5  8 11 16 18 27], pred: [ 3  6 14 22 25 32]
2023-03-20 11:45:54.064 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 16.263904571533203, tag: [ 7 14 15 19 21 28], pred: [ 1  7 14 22 26 32]
2023-03-20 11:45:54.834 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 15.361801147460938, tag: [ 8 14 22 24 27 29], pred: [ 1  7 10 17 25 32]
2023-03-20 11:45:55.602 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 18.51010513305664, tag: [ 9 23 24 25 29 31], pred: [ 1  7 16 19 29 33]
2023-03-20 11:45:56.409 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 14.091325759887695, tag: [ 1 14 15 17 26 30], pred: [ 2  4 13 19 23 33]
2023-03-20 11:45:57.372 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 17.22756576538086, tag: [ 3  7 10 13 14 25], pred: [ 1  7 17 22 28 32]
2023-03-20 11:45:58.237 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 16.53325080871582, tag: [12 15 16 26 29 31], pred: [ 2  4 13 19 27 33]
2023-03-20 11:45:59.067 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 16.074129104614258, tag: [ 5 13 15 17 19 21], pred: [ 3  6 11 17 27 32]
2023-03-20 11:45:59.899 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 15.145319938659668, tag: [ 9 11 16 17 25 27], pred: [ 1  7 18 20 26 29]
2023-03-20 11:46:00.743 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 13.291736602783203, tag: [ 1  3 16 18 22 29], pred: [ 1  7  8 17 25 30]
2023-03-20 11:46:01.909 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 14.11507511138916, tag: [ 9 12 14 20 30 31], pred: [ 1  7 18 20 25 30]
2023-03-20 11:46:02.899 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 14.175811767578125, tag: [ 1 12 16 18 22 33], pred: [ 1  7 10 15 26 30]
2023-03-20 11:46:03.750 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 16.240161895751953, tag: [ 6  8 11 13 17 19], pred: [ 1  7 18 20 26 30]
2023-03-20 11:46:04.584 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 17.57523536682129, tag: [14 17 21 29 31 32], pred: [ 3  5 12 18 30 32]
2023-03-20 11:46:05.548 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 13.9828519821167, tag: [ 9 12 14 18 27 33], pred: [ 1  3 18 20 31 32]
2023-03-20 11:46:06.898 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 14.383771896362305, tag: [ 4  6 12 19 27 29], pred: [ 3 16 18 20 31 32]
2023-03-20 11:46:07.915 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 14.403094291687012, tag: [ 2 11 13 18 22 30], pred: [ 1  7 18 20 26 30]
2023-03-20 11:46:08.949 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 14.503276824951172, tag: [ 3  8 11 29 30 32], pred: [ 1  7 18 20 31 32]
2023-03-20 11:46:10.342 | INFO     | __main__:train_red_ball_model:100 - epoch: 0, loss: 13.981236457824707, tag: [ 2 12 19 22 27 32], pred: [ 1  7 18 20 25 30]
WARNING:tensorflow:Issue encountered when serializing variables.
Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.
'transitions' has type str, but expected one of: int, long, bool
WARNING:tensorflow:Issue encountered when serializing trainable_variables.
Type is unsupported, or the types of the items don't match field type in CollectionDef. Note this is a warning and probably safe to ignore.
'transitions' has type str, but expected one of: int, long, bool
2023-03-20 11:46:12.583 | INFO     | __main__:run:199 - 训练耗时: 42.83347249031067
2023-03-20 11:46:12.585 | INFO     | __main__:run:203 - 开始训练【双色球】蓝球模型...
2023-03-20 11:46:12.587 | INFO     | __main__:train_blue_ball_model:122 - 特征数据维度: (2970, 3)
2023-03-20 11:46:12.587 | INFO     | __main__:train_blue_ball_model:123 - 标签数据维度: (2970, 16)
2023-03-20 11:46:15.994 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.7723209857940674, tag: 6, pred: 12
2023-03-20 11:46:16.459 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.7735366821289062, tag: 4, pred: 6
2023-03-20 11:46:16.786 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.673487901687622, tag: 6, pred: 14
2023-03-20 11:46:17.203 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.7841591835021973, tag: 5, pred: 15
2023-03-20 11:46:17.620 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.68033766746521, tag: 1, pred: 15
2023-03-20 11:46:18.042 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.797462224960327, tag: 5, pred: 15
2023-03-20 11:46:18.438 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.753267526626587, tag: 4, pred: 1
2023-03-20 11:46:18.853 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.654209852218628, tag: 1, pred: 1
2023-03-20 11:46:19.294 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.7605557441711426, tag: 15, pred: 1
2023-03-20 11:46:19.730 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.773200750350952, tag: 15, pred: 16
2023-03-20 11:46:20.169 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.719604969024658, tag: 7, pred: 7
2023-03-20 11:46:20.612 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.7519779205322266, tag: 4, pred: 12
2023-03-20 11:46:20.963 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.706936836242676, tag: 7, pred: 12
2023-03-20 11:46:21.263 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.797590732574463, tag: 10, pred: 12
2023-03-20 11:46:21.567 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.6708242893218994, tag: 12, pred: 12
2023-03-20 11:46:22.058 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.845566511154175, tag: 2, pred: 12
2023-03-20 11:46:22.436 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.781891345977783, tag: 11, pred: 7
2023-03-20 11:46:22.744 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.864372968673706, tag: 2, pred: 12
2023-03-20 11:46:23.047 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.74739146232605, tag: 15, pred: 12
2023-03-20 11:46:23.354 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.8099265098571777, tag: 14, pred: 16
2023-03-20 11:46:23.650 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.820903778076172, tag: 4, pred: 12
2023-03-20 11:46:23.951 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.749382972717285, tag: 6, pred: 12
2023-03-20 11:46:24.289 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.7867088317871094, tag: 4, pred: 6
2023-03-20 11:46:24.630 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.7657570838928223, tag: 12, pred: 1
2023-03-20 11:46:25.021 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.7908122539520264, tag: 12, pred: 1
2023-03-20 11:46:25.452 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.8512625694274902, tag: 13, pred: 1
2023-03-20 11:46:25.767 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.7639896869659424, tag: 8, pred: 1
2023-03-20 11:46:26.065 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.818244457244873, tag: 4, pred: 9
2023-03-20 11:46:26.432 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.7372004985809326, tag: 1, pred: 9
2023-03-20 11:46:26.952 | INFO     | __main__:train_blue_ball_model:166 - epoch: 0, loss: 2.745108127593994, tag: 1, pred: 14
2023-03-20 11:46:28.046 | INFO     | __main__:run:206 - 训练耗时: 15.459038972854614
